spec/features/publish_daily_statistics_spec.rb

```diff
@@ -0,0 +1,74 @@
+describe 'synchronizing IPs and locations' do
+  let(:ip_locations) { DB[:ip_locations] }
+
+  before do
+    ENV['S3_PUBLISHED_LOCATIONS_IPS_BUCKET'] = 'stub-bucket'
+    ENV['S3_PUBLISHED_LOCATIONS_IPS_OBJECT_KEY'] = 'stub-key'
+
+    stub_request(:get, "http://169.254.169.254/latest/meta-data/iam/security-credentials/")
+      .to_return(body: '')
+
+    stub_request(:get, 'https://stub-bucket.s3.eu-west-2.amazonaws.com/stub-key') \
+      .to_return(body: object_content.to_json)
+
+    ip_locations.truncate
+  end
+
+  subject do
+    source_gateway = PerformancePlatform::Gateway::S3IpLocations.new
```

Comment:
It would be better to run the rake task directly here, rather than replicating its contents, but testing the rake task cost us an afternoon - this is the best we've got for now.